pkg/controller/installation/core_controller.go

```diff
@@ -1072,11 +1076,8 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 		typhaNodeTLS.CAConfigMap,
 	}
 
-	if typhaNodeTLS.NodeSecret != nil {
-		objs = append(objs, typhaNodeTLS.NodeSecret)
-	}
-	if typhaNodeTLS.TyphaSecret != nil {
-		objs = append(objs, typhaNodeTLS.TyphaSecret)
+	if renderNodeAndTyphaSecrets {
+		objs = append(objs, typhaNodeTLS.NodeSecret, typhaNodeTLS.TyphaSecret)
```

Comment:
The typhaNodeTLS.CAConfigMap should be treated the same as the Node and Typha secrets when they are created. The difference is that if using CertificateManagement then we will also need to render the CAConfigMap.
I'd suggest creating `objs = []client.Object{}` at Line 919 and then instead of setting `renderNodeAndTyphaSecrets`, you can just add the appropriate ConfigMap and Secrets to `objs` when they are created. That way you don't need to create a variable (or more) to pass the need to create/render them to a different part of the code.

Reply:
`objs` definition moved up and objects are now appended following their creation